synapse/events/snapshot.py

```diff
@@ -198,6 +201,11 @@ def deserialize(storage: "Storage", input: JsonDict) -> "EventContext":
             rejected=input["rejected"],
         )
 
+        if input["outlier"]:
+            context._prev_state_ids = {}
+            context._current_state_ids = {}
```

Comment:
why are these necessary? shouldn't they be `{}` for an outlier anyway?

Reply:
The problem is that `serialize` doesn't preserve those private fields, so they become `None` when transmitted to an event persistence worker. Being `None` means something will try to re-populate the values, which will fail because fetching those values for outlier events is not possible.

Reply:
Right, I see.

Let's do what @MadLittleMods suggests and have`_AsyncEventContextImpl._fill_out_state `to default current_state_ids={} and prev_state_ids={} if `state_group is None`

Reply:
closing this thread in favour of https://github.com/matrix-org/synapse/pull/11220/files#r747812503